src/shared/utilities/collectionUtils.ts

```diff
@@ -272,3 +272,7 @@ export function stripUndefined(obj: any): void {
         }
     })
 }
+
+export function isAsyncIterable(obj: any): obj is AsyncIterable<unknown> {
+    return Object.getOwnPropertySymbols(obj).includes(Symbol.asyncIterator)
```

Comment:
I don't think this only scopes down to `AsyncIterable`s. I would be wary if you're using this to 100% determine whether or not something is an AsyncIterable.

Reply:
By definition (at least from the interface) an `AsyncIterable` only has a symbol property that returns a function to create an `AsyncIterator`. So yes this would technically cover other things like an `AsyncGenerator` but that's intended. I suppose it'd be wise to check that the property is a function as well.

Reply:
I know you're on the warpath against `AsyncIterableIterator`s and I'm not sure if this is a part of that, but those have [`Symbol.asyncIterator` by design](https://devblogs.microsoft.com/azure-sdk/async-iterators-in-the-azure-sdk-for-javascript-typescript/).

Reply:
Yes I know. It's all the same abstraction, just typed differently. `AsyncIterableIterator` _is_ an `AsyncIterable`, just the former is delegating to another iterator. In the end consumers don't care about this, it all looks the same.